org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/consistencycheck/ConsistencyCheck.java

```diff
@@ -1,23 +1,146 @@
+
+		errors.addAll(checkConsistencyArtifactTemplateId(repository.getAllDefinitionsChildIds(ArtifactTemplateId.class), repository));
+		errors.addAll(checkConsistencyPolicyTemplateId(repository.getAllDefinitionsChildIds(PolicyTemplateId.class), repository));
+		errors.addAll(checkConsistencyServiceTemplateId(repository.getAllDefinitionsChildIds(ServiceTemplateId.class), repository));
+		
+		errors.addAll(checkConsistencyNodeTypeImplementationId(repository.getAllDefinitionsChildIds(NodeTypeImplementationId.class), repository));
+		errors.addAll(checkConsistencyRelationshipTypeImplementationId(repository.getAllDefinitionsChildIds(RelationshipTypeImplementationId.class), repository));
+		errors.addAll(checkConsistencyArtifactTypeId(repository.getAllDefinitionsChildIds(ArtifactTypeId.class), repository));
+		
+		errors.addAll(checkConsistencyCapabilityTypeId(repository.getAllDefinitionsChildIds(CapabilityTypeId.class), repository));
+		errors.addAll(checkConsistencyNodeTypeId(repository.getAllDefinitionsChildIds(NodeTypeId.class), repository));
+		errors.addAll(checkConsistencyRequirementTypeId(repository.getAllDefinitionsChildIds(RequirementTypeId.class), repository));
+
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyArtifactTemplateId(Set<ArtifactTemplateId> checkForLinkedInRepository, IRepository repository) {
+		// Get All ArtifactTemplateId from Repository 
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (ArtifactTemplateId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyPolicyTemplateId(Set<PolicyTemplateId> checkForLinkedInRepository, IRepository repository) {
+		// Get All PolicyTemplateId from Repository 
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (PolicyTemplateId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyServiceTemplateId(Set<ServiceTemplateId> checkForLinkedInRepository, IRepository repository) {
+		// Get All ServiceTemplateId from Repository 
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (ServiceTemplateId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyNodeTypeImplementationId(Set<NodeTypeImplementationId> checkForLinkedInRepository, IRepository repository) {
+		// Get All NodeTypeImplementationId from Repository 
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (NodeTypeImplementationId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyRelationshipTypeImplementationId(Set<RelationshipTypeImplementationId> checkForLinkedInRepository, IRepository repository) {
```

Comment:
Do not name the method `...RelationshipTypeImplementationId`, but only `...RelationshipTypeImplementation`. Thus, leaving out `Id`. Similar for all other methods.